src/ConstitutiveRelationships.jl

```diff
@@ -17,30 +17,48 @@ using StaticArrays
 
 const AxialCompression, SimpleShear, Invariant = 1, 2, 3
 
-abstract type AbstractConstitutiveLaw{T} <: AbstractMaterialParam end
+#abstract type AbstractConstitutiveLaw{T} <: AbstractMaterialParam end
+#abstract type AbstractComposite <: AbstractMaterialParam end
+
+precision(v::AbstractConstitutiveLaw) = (typeof(v).super).parameters[1]
```

Comment:
Can be simplified as `precision(v::AbstractConstitutiveLaw{T})  where T= T`

Reply:
yes thanks